src/test/kotlin/org/rust/ide/hints/RsInlayParameterHintsProviderTest.kt

```diff
@@ -60,7 +60,14 @@ class RsInlayParameterHintsProviderTest : RsTestBase() {
         fn main() {
             let s/*caret*/ = S;
         }  //^
-    """, ": S", 0)
+    """, ": S", 0, smart = false)
+
+    fun `test let should not show a hint directly after initial struct smart hint`() = checkNoHint<RsLetDecl>("""
```

Comment:
Let's add a couple of more tests here, where the hints should not be shown?

I am thinking about

```Rust
struct TupleStruct(f32)
struct BracedStruct { f: f32 }
enum E {
   C, B { f: f32 }, T(f32)
}

fn main() {
    let no_hint = TupleStruct(1.0);
    let no_hint = BracedStruct { f: 1.0 };
    let no_hint = E::C;
    let no_hint = E::B { f: 1.0 };
    let no_hint = E::T(1.0);
}
```